spec/system/admin/reports/sales_tax/sales_tax_totals_by_order_spec.rb

```diff
@@ -0,0 +1,433 @@
+  let!(:table_header){
+    [
+      "Distributor",
+      "Order Cycle",
+      "Order ID",
+      "Tax Category",
+      "Tax Rate Name",
+      "Tax Rate",
+      "Total excl. Tax ($)",
+      "Tax",
+      "Total incl. Tax ($)",
+      "First Name",
+      "Last Name",
+      "Code",
+      "Email"
+    ].join(" ").upcase
+  }
+  let!(:state_zone){ create(:zone_with_state_member) }
+  let!(:country_zone){ create(:zone_with_member) }
+  let!(:tax_category){ create(:tax_category, name: 'tax_category') }
+  let!(:state_tax_rate){
+    create(:tax_rate, zone: state_zone, tax_category: tax_category,
+                      name: 'State', amount: 0.015)
+  }
+  let!(:country_tax_rate){
+    create(:tax_rate, zone: country_zone, tax_category: tax_category,
+                      name: 'Country', amount: 0.025)
+  }
+  let!(:ship_address){
+    create(:ship_address,
+           state: state_zone.members.first.zoneable,
+           country: country_zone.members.first.zoneable)
+  }
+
+  let!(:variant){ create(:variant) }
+  let!(:product){ variant.product }
+  let!(:supplier){ create(:supplier_enterprise, name: 'Supplier', charges_sales_tax: true) }
+  let!(:distributor){
+    create(:distributor_enterprise_with_tax, name: 'Distributor', charges_sales_tax: true)
+  }
+  let!(:distributor_fee){
+    create(:enterprise_fee, :flat_rate, amount: 5,
+                                        tax_category_id: tax_category.id,
+                                        enterprise_id: distributor.id)
+  }
+  let!(:payment_method){ create(:payment_method, :flat_rate) }
+  let!(:shipping_method){
+    create(:shipping_method, :flat_rate, amount: 10, tax_category_id: tax_category.id)
+  }
+
+  let!(:order){ create(:order_with_distributor, distributor: distributor) }
+  let!(:order_cycle){
+    create(:simple_order_cycle, name: 'oc1', suppliers: [supplier], distributors: [distributor],
+                                variants: [variant])
+  }
+  let!(:customer1){
+    create(:customer, enterprise: create(:enterprise),
+                      user: create(:user),
+                      first_name: 'cfname', last_name: 'clname', code: 'ABC123')
+  }
+
+  let(:admin){ create(:admin_user) }
+
+  before do
+    order_cycle.cached_outgoing_exchanges.first.enterprise_fees << distributor_fee
+    distributor.shipping_methods << shipping_method
+    distributor.payment_methods << payment_method
+
+    product.update!({
+                      tax_category_id: tax_category.id,
+                      supplier_id: supplier.id
+                    })
+
+    order.update!({
+                    number: 'ORDER_NUMBER_1',
+                    order_cycle_id: order_cycle.id,
+                    ship_address_id: ship_address.id,
+                    customer_id: customer1.id,
+                    email: 'order1@example.com'
+                  })
+    order.line_items.create({ variant: variant, quantity: 1, price: 100 })
+  end
+  context 'added tax' do
```

Comment:
I'm surprised we don't have a linter rule for this, but I think each block should have an empty line separating them.

```suggestion
  end
  
  context 'added tax' do
```